npm/test/upload-form-files.test.ts

```diff
@@ -0,0 +1,565 @@
+  }
+
+  setUpFileUploader(): FileUploader {
+    const mockUpdateToken = jest.fn()
+    const isTokenExpired = true
+    const refreshTokenParsed: KeycloakTokenParsed = {
+      exp: Math.round(new Date().getTime() / 1000) + 60
+    }
+    const isJudgmentUser: boolean = true
+    const mockKeycloakInstance: KeycloakInstance = createMockKeycloakInstance(
+      mockUpdateToken,
+      isTokenExpired,
+      refreshTokenParsed,
+      isJudgmentUser
+    )
+    const client = new GraphqlClient(
+      "https://example.com",
+      mockKeycloakInstance
+    )
+    const frontendInfo: IFrontEndInfo = {
+      apiUrl: "",
+      region: "",
+      stage: "test",
+      uploadUrl: ""
+    }
+    const uploadMetadata = new ClientFileMetadataUpload(client)
+    const updateConsignmentStatus = new UpdateConsignmentStatus(client)
+    return new FileUploader(
+      uploadMetadata,
+      updateConsignmentStatus,
+      frontendInfo,
+      mockGoToNextPage,
+      mockKeycloakInstance
+    )
+  }
+
+  uploadYourRecordsSection: HTMLElement | null =
+    document.querySelector("#file-upload")
+  dropzone: HTMLElement | null = document.querySelector(
+    ".drag-and-drop__dropzone"
+  )
+  uploadForm: HTMLFormElement | null =
+    document.querySelector("#file-upload-form")
+  fileRetriever: HTMLInputElement | null =
+    document.querySelector("#file-selection")
+
+  fileRetrievalSuccessMessage: HTMLElement | null = document.querySelector(
+    ".drag-and-drop__success"
+  )
+
+  fileNameElement: HTMLElement | null = document.querySelector("#file-name")
+
+  warningMessages: {
+    [s: string]: HTMLElement | null
+  } = {
+    incorrectItemSelectedMessage: document.querySelector(
+      "#item-selection-failure"
+    ),
+    submissionWithoutSelectionMessage: document.querySelector(
+      "#nothing-selected-submission-message"
+    )
+  }
+
+  warningMessagesText: {
+    [s: string]: HTMLElement | null
+  } = {
+    incorrectItemSelectedMessageText: document.querySelector(
+      "#non-file-selected-message-text"
+    )
+  }
+
+  hiddenInputButton: HTMLElement | null =
+    document.querySelector("#file-selection")
+
+  submitButton: HTMLElement | null = document.querySelector(
+    "#start-upload-button"
+  )
+
+  fileUploader = this.setUpFileUploader()
+
+  form = new UploadForm(
+    true,
+    this.uploadForm!,
+    this.fileRetriever!,
+    this.dropzone!,
+    this.setUpFileUploader
+  )
+
+  uploadingRecordsSection = document.querySelector("#upload-progress")
+}
+
+test("clicking the submit button, without selecting a file, doesn't reveal the progress bar", async () => {
+  const mockDom = new MockDom()
+
+  const submitEvent = mockDom.createSubmitEvent()
+  await mockDom.form.handleFormSubmission(submitEvent)
+
+  expect(mockDom.uploadingRecordsSection).toHaveAttribute("hidden")
+})
+
+test("clicking the submit button, without selecting a file, doesn't disable the buttons on the page", async () => {
```

Comment:
Could the "_clicking the submit button, without selecting a file ..._" tests just be combined into a single test, and just check all the different elements in the single test?

Reply:
Ah, yes, that's a good idea. I'll do the same for the upload page tests too